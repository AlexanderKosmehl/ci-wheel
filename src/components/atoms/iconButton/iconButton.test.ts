import {
  describe, expect, it, vi,
} from 'vitest';
import generateIconButton from './iconButton';
import addIconUrl from '../../../assets/+-Icon.svg';

describe('iconButton', () => {
  const onClick = vi.fn(() => {});

  const button = generateIconButton({
    iconURL: addIconUrl,
    classes: ['class', 'class2'],
    onClick,
  });

  it('generates the component', () => {
    expect(button).toMatchSnapshot();
  });

  it('adds working clickListener', () => {
    button.click();
    expect(onClick).toBeCalled();
  });
});
