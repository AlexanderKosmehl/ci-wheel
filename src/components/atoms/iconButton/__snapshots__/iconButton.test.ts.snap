// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iconButton generates the component 1`] = `
<button
  class="button class class2"
>
  <img
    class="icon"
    src="test-file-stub"
  />
</button>
`;
